beginner_source/onnx/onnx_diagnostics_tutorial.py

```diff
@@ -0,0 +1,181 @@
+# -*- coding: utf-8 -*-
+
+"""
+`Introduction to ONNX <intro_onnx.html>`_ ||
+`Export a PyTorch model to ONNX <export_simple_model_to_onnx_tutorial.html>`_ ||
+`Introduction to ONNX Registry <onnx_registry_tutorial.html>`_ ||
+**Introduction to ONNX Diagnostics**
+
+Introduction to ONNX Diagnostics
+================================
+
+**Author:** Bowen Bao (bowbao@microsoft.com)
+"""
+
+###############################################################################
+# Overview
+# --------
+#
+# Welcome to this tutorial on ONNX Diagnostics. ONNX diagnostics goes beyond regular logs through the adoption of
+# `Static Analysis Results Interchange Format (aka SARIF) <https://docs.oasis-open.org/sarif/sarif/v2.1.0/sarif-v2.1.0.html>`__
+# to help users debug and improve their model using a GUI, such as
+# Visual Studio Code's `SARIF Viewer <https://marketplace.visualstudio.com/items?itemName=MS-SarifVSCode.sarif-viewer>`_,
+# or `online SARIF Viewer <https://microsoft.github.io/sarif-web-component/>`_, etc.
+# Each diagnostic rule is documented at `ONNX Diagnostic Rules <https://pytorch.org/docs/main/onnx_dynamo.html#diagnosing-issues-with-sarif>`_.
```

Comment:
```suggestion
# The new ONNX exporter introduced a Diagnostic component that goes beyond regular logs through the adoption of
# `Static Analysis Results Interchange Format (aka SARIF) <https://docs.oasis-open.org/sarif/sarif/v2.1.0/sarif-v2.1.0.html>`__. 
# In a nutshell, SARIF serves as a structured and consistent means of representing the outcomes of the ONNX export process.
# It offers a comprehensive framework that includes information about analysis runs, results, rules, and tool metadata. 
# At its core, SARIF is organized hierarchically, starting with a root object that encapsulates the entire document.
# Within this structure, it defines key components such as runs (representing a specific onnx export run),
# results (individual issue findings), and rules (the issue categories that the exporter has hit).
# With all this infrastructure in place, users can run, debug and improve their model using a GUI, such as
# Visual Studio Code's `SARIF Viewer <https://marketplace.visualstudio.com/items?itemName=MS-SarifVSCode.sarif-viewer>`_,
# or `online SARIF Viewer <https://microsoft.github.io/sarif-web-component/>`_, etc.
# The diagnostic rules serve as a predefined set of issues categories reported during the export process and are documented at `ONNX Diagnostic Rules <https://pytorch.org/docs/main/onnx_dynamo.html#diagnosing-issues-with-sarif>`_.
```

How about building a little on what SARIF is, how it works internally and then jump on the practical use in the exporter? It helps new users to create a mental map on what SARIF is, how it improves regular logging and helps understanding the role of "rules" and why we care about it

ps: I didn't test the rendering of that extra paragraph